source/reference/best-practices.rst

```diff
@@ -47,11 +92,19 @@ Some application frameworks such as Qt have this functionality built-in, for exa
    For an example, see: https://github.com/KaidanIM/Kaidan/commit/da38011b55a1aa5d17764647ecd699deb4be437f
 
 .. warning::
-   
+
    :code:`QStandardPaths::standardLocations(QStandardPaths::AppDataLocation)` **does not work reliably.**
-   
+
    According to the `Qt documentation`_, this resolves to :code:`~/.local/share/<APPNAME>`, :code:`/usr/local/share/<APPNAME>`, :code:`/usr/share/<APPNAME>`, but clearly :code:`/usr` is not where these things are located in an AppImage.
 
+..
+  I've added this. Should it be removed?
+
+If for some reason you're unable to get your appimage working with
```

Comment:
What do you think about this paragraph I added?

Reply:
You wanted to mention `$APPDIR` here, which points to the runtime mountpoint.

It's a good idea to put this on the bottom of the page, since this approach has significant drawbacks. Normally, this environment variable is *only* available when an application is actually run via the AppImage runtime. The environment variable would only be available in an extracted AppImage or an AppDir right before packaging if there is a suitable `AppRun` (script) that defines it.

Should we include an example script?

Reply:
> You wanted to mention `$APPDIR` here, which points to the runtime mountpoint.

Ah, I see that APPDIR would be right, not APPIMAGE for this case.

> 
> It's a good idea to put this on the bottom of the page, since this approach has significant drawbacks. Normally, this environment variable is _only_ available when an application is actually run via the AppImage runtime. The environment variable would only be available in an extracted AppImage or an AppDir right before packaging if there is a suitable `AppRun` (script) that defines it.

I'm a bit confused. I'm interpreting this as two different things. 1. It's available by default during runtime of any appimage; 2. It's only available in an extracted appimage before packaging and only if there's a suitable AppRun script that defines it.

I've just tried using APPDIR with a small program I maintain. I inserted this code into main(), at the top:

```c
  char *appdir = getenv("APPDIR");
  if (appdir != NULL)
    printf("%s\n", appdir);
```
It prints out the mountpoint. I'm not using an AppRun script, just a symlink. So my thinking is that if a developer had trouble using relatives paths to data for example, they could simply do something like:

```pseudo
if getenv APPDIR == NULL
  use unmodified data path
else
  data = $APPDIR/data...
```

And as far as I can tell, that would be robust. Please correct me if I'm missing something.

/cc @jammyjamjamman